### ISSUE TITLE
Should error if macro not closed

### ISSUE THREAD

**Issue by PabstMirror** [BrettMayson/HEMTT#548]
```
#ifdef A
    #ifdef B
    x = 1;
    #else
    x = 2;
#else
    #ifdef B
    x = 3;
    #else
    x = 4;
#endif
```
subtle error in this code, but hemtt will build with no `x=` present, 
should probably error if `#if` macros aren't closed

**Comment by BrettMayson** [comment]
Should also be fixed to not allow
```
#else
#else
```
as that really isn't valid

**Comment by BrettMayson** [comment]
Implemented in #556
